runtime/gc_vlhgc/InterRegionRememberedSet.hpp

```diff
@@ -185,8 +189,39 @@ class MM_InterRegionRememberedSet : public MM_BaseVirtual
 	void clearFromRegionReferencesForCompactOptimized(MM_EnvironmentVLHGC* env);
 	MM_InterRegionRememberedSet(MM_HeapRegionManager *heapRegionManager);
 	
+
+	/**
+	 * Helper function isCompressedCardDirtyForPartialCollect
+	 * extended from compressedCardTable->isCompressedCardDirtyForPartialCollect(), only in case first PGC after GMP, also dirty card if card Contains No Object.
+	 */
+	MMINLINE bool isCompressedCardDirtyForPartialCollect(MM_EnvironmentVLHGC* env, UDATA card, MM_CompressedCardTable *compressedCardTable, bool needToCheckMarkMap, MM_MarkMap *markMap)
+	{
+		void* heapAddress = convertHeapAddressFromRememberedSetCard(card);
+		bool ret = compressedCardTable->isCompressedCardDirtyForPartialCollect(env, heapAddress);
+		if (!ret && needToCheckMarkMap) {
+			ret = markMap->areAnyLiveObjectsInCard(heapAddress);
+		}
+		return ret;
+
+	}
 protected:
 public:
+	/**
+	 * Helper function cardMayContainObjects
+	 * if this is first PGC after GMP, check if markMap->areAnyLiveObjectsInCard()
+	 * otherwise check if region->containsObjects()
+	 */
+	MMINLINE bool cardMayContainObjects(UDATA card, MM_HeapRegionDescriptorVLHGC *fromRegion, bool needToCheckMarkMap, MM_MarkMap *markMap)
```

Comment:
both helper could by static (hence saving by not passing implicit 'this' pointer)